recipes-bsp/u-boot/u-boot-distro-boot/uEnv.txt.in

```diff
@@ -108,6 +112,116 @@ bootcmd_boot=if test ${bootscript_debug} != 1; then \
                  fi; \
              fi || true
 
-bootcmd_run=run board_fixups && run check_rollback_needed && run set_bootargs && run set_fdt_path && \
+prog_fuses=if test -n "${fuse_prog_list}"; then \
+               for fuse_val in "${fuse_prog_list}"; do \
+                   if test "${fuse_fail}" != "1"; then \
+                       if test "${fuse_dry_run}" = "1"; then \
+                           echo "DRY RUN: fuse prog ${fuse_bank} ${fuse_word} ${fuse_val}"; \
+                       else \
+                           fuse prog -y ${fuse_bank} ${fuse_word} ${fuse_val}; \
+                       fi; \
+                       if test $? != "0"; then \
+                           env set fuse_fail 1; \
+                       fi; \
+                       if test "${fuse_dry_run}" = "1"; then \
+                           echo "DRY RUN: fuse cmp"; \
+                       else \
+                           fuse cmp ${fuse_bank} ${fuse_word} ${fuse_val}; \
+                       fi; \
+                       if test $? != "0"; then \
+                           env set fuse_fail 1; \
+                       fi; \
+                   fi ; \
+                   setexpr fuse_word $fuse_word + 1; \
+                   if test "${soc}" = "imx8m" || test "${soc}" = "imx7d"; then \
+                       if test "${fuse_word}" = "4"; then \
+                           setexpr fuse_bank $fuse_bank + 1; \
+                           setexpr fuse_word 0; \
+                       fi; \
+                   fi; \
+                   if test "${fuse_word}" = "72a"; then \
+                       env set fuse_word 730; \
+                   elif test "${fuse_word}" = "73a"; then \
+                       env set fuse_word 740; \
+                   fi; \
```

Comment:
I see you created a couple of machine specific variables to define the fuse bank/word, fuse commands and so on all configurable by bitbake variables which is great. However, here we have this code that is machine specific and it takes runtime decisions based on the soc which means the "knowledge" about the machine is now spread into two places which does not look ideal.

So I was wondering: couldn't we make this script more machine agnostic?

One way I imagine doing this is by having preset variables already with the bank/word pairs (one pair for each element in `FUSE_NUM`), e.g.:

```
fuse_bank_word1=0 722
fuse_bank_word2=1 723
fuse_bank_word3=2 724
fuse_bank_word4=3 725
[...]
```

Then we could loop over the variables and drop this machine specific logic. Plus, the code would be shorter.

What do you think?

Reply:
I considered doing this. I just didn't want to hard-code 8/16 fuse/bank values in variables, I thought it looked "ugly" so to speak.

I used how we generate the `fuse-cmds.txt` file in `meta-toradex-security` as an inspiration for this logic. If you think it's better to just have the hard-coded variables I can change this.

Reply:
Done in V2. Just went with a hard-coded list of fuse-bank variables.

Reply:
Ok, I have some comments/suggestions regarding this, but I'll resolve this thread and add them on my review for V2.